### PR TITLE
Add results tables

### PR DESCRIPTION
**Pull request by iasonsky**

### CODE REVIEW

File: EquiCLIP/main_lambda_equitune.py [hunk 1]
```diff
@@ -103,13 +103,19 @@ def main(args):
...
+                )
+                wandb.log({"prefinetune_top1_acc": prefinetune_top1_acc, "prefinetune_top5_acc": prefinetune_top5_acc, "prefinetune_precision": prefinetune_precision, 
+                        "prefinetune_recall": prefinetune_recall, "prefinetune_f1_score": prefinetune_f1_score})
+            else:
+                # evaluating for only 50 steps using val=True
+                prefinetune_top1_acc, prefinetune_top5_acc, prefinetune_precision, prefinetune_recall, prefinetune_f1_score = eval_clip(
+                    args, model, zeroshot_weights, train_loader, val=True, **val_kwargs
+                )
+                wandb.log({"prefinetune_top1_acc": prefinetune_top1_acc, "prefinetune_top5_acc": prefinetune_top5_acc, "prefinetune_precision": prefinetune_precision, 
+                        "prefinetune_recall": prefinetune_recall, "prefinetune_f1_score": prefinetune_f1_score})
```
Comment (mikhail-vlasenko) [inline EquiCLIP/main_lambda_equitune.py:118]: avoid code duplication

File: EquiCLIP/main_lambda_equitune.py [hunk 2]
```diff
@@ -124,9 +130,13 @@ def main(args):
...
-    # val=True only for choosing the best lambda weights using the trainloader
-    val_top1_acc, val_top5_acc, val_precision, val_recall, val_f1_score = eval_clip(args, model, zeroshot_weights, eval_loader, val=True, **val_kwargs)
-    wandb.log({"val_top1_acc": val_top1_acc, "val_top5_acc": val_top5_acc, "val_precision": val_precision, "val_recall": val_recall, "val_f1_score": val_f1_score})
+    if args.full_val:
+        val_top1_acc, val_top5_acc, val_precision, val_recall, val_f1_score = eval_clip(args, model, zeroshot_weights, eval_loader, val=False, **val_kwargs)
+        wandb.log({"val_top1_acc": val_top1_acc, "val_top5_acc": val_top5_acc, "val_precision": val_precision, "val_recall": val_recall, "val_f1_score": val_f1_score})
+    else:
+        # val=True only for choosing the best lambda weights using the trainloader
+        val_top1_acc, val_top5_acc, val_precision, val_recall, val_f1_score = eval_clip(args, model, zeroshot_weights, eval_loader, val=True, **val_kwargs)
+        wandb.log({"val_top1_acc": val_top1_acc, "val_top5_acc": val_top5_acc, "val_precision": val_precision, "val_recall": val_recall, "val_f1_score": val_f1_score})
```
Comment (mikhail-vlasenko) [inline EquiCLIP/main_lambda_equitune.py:139]: code duplication

File: EquiCLIP/weighted_equitune_utils.py [hunk 3]
```diff
@@ -124,16 +118,9 @@ def weighted_equitune_clip(args, model: CLIP,
...
-    import time
     st_time = time.time()
-    for i in trange(num_iterations, desc="Training CLIP and/or WeightNet"):
-        # if (i+1)%iter_print_freq == 0:
-        #     print(f"iteration number: {i+1}")
-        #     curr_time = time.time()
-        #     print(f"time elapsed per iter: {(curr_time - st_time) / (i + 1)}")
-        (images, target) = next(training_iterator)
+    limited_loader = islice(loader, num_iterations)
+    for i, (images, target) in enumerate(tqdm(limited_loader, desc="Training CLIP and/or WeightNet", total=num_iterations)):
```
Comment (mikhail-vlasenko) [inline EquiCLIP/weighted_equitune_utils.py:123]: `i` is unused

**Review by mikhail-vlasenko — CHANGES_REQUESTED** [review]

**Comment by iasonsky** [comment]
Made the changes you asked, but I don't want to merge yet, cause then I won't be able to run any experiments on the "original" code.

File: EquiCLIP/main_lambda_equitune.py (hunk 1 above)
Comment (iasonsky) [reply EquiCLIP/main_lambda_equitune.py]: did the change, but might as well skip those arguments altogether and stick with one setting like main.

**Comment by mikhail-vlasenko** [comment]
lets just make a branch for the original code. we can also merge new changes there when we need to

File: EquiCLIP/main_lambda_equitune.py [hunk 4]
```diff
@@ -105,13 +105,12 @@ def main(args):
                 print(f"Learning lambda weights: {i}/{args.num_prefinetunes}")
             # zeroshot prediction
             # add weight_net save code for the best model
-
-            # evaluating for only 50 steps using val=True
+            val = False if args.full_val_pf else True # evaluating for only 50 steps using val=True if full_val_pf is False
```
Comment (iasonsky) [inline EquiCLIP/main_lambda_equitune.py:108]: maybe we can omit that and stick with val=True?

**Review by iasonsky — COMMENTED** [review]
I think it's ready now. Check my comment on `main_lambda_equitune.py` line 108.

File: EquiCLIP/main_lambda_equitune.py (hunk 4 above)
Comment (mikhail-vlasenko) [reply EquiCLIP/main_lambda_equitune.py:108]: i dont mind

**Review by mikhail-vlasenko — APPROVED** [review]
lgtm

**Pull request merged**